### PR TITLE
Check for DLC presence in one spot

### PR DESCRIPTION
**Pull request by luckysori**
Actually fixes what https://github.com/itchysats/itchysats/pull/2835 set out to fix.

Patch e39a0134950d92931a8534d7d8c3f4380d908e0c was a step in the right direction in terms of suppressing false positives about not being able to auto-rollover CFDs without a DLC, it wasn't enough.

It was insufficient because we were checking for DLC presence in more than one spot in the same method. By checking in only one spot we can simplify the method and also ensure that we check if the CFD is empty _before_ we check for DLC presence.

### CODE REVIEW

**Comment by luckysori** [comment]
Tested locally with DB with empty CFDs.

File: model/src/cfd.rs [hunk 1]
```diff
@@ -704,16 +698,19 @@ impl Cfd {
         &self,
         now: OffsetDateTime,
     ) -> Result<(Txid, BitMexPriceEventId), CannotRollover> {
-        let expiry_timestamp = self.expiry_timestamp().ok_or(CannotRollover::NoDlc)?;
+        self.can_rollover()?;
+
+        let dlc = match &self.dlc {
```
Comment (da-kami) [inline model/src/cfd.rs]: ❓Any particular reason why you prefer the `match` over the `ok_or`? I think it should be equivalent.

**Review by da-kami — APPROVED** [review]

File: model/src/cfd.rs [hunk 2]
```diff
@@ -704,16 +698,19 @@ impl Cfd {
...
+            Some(dlc) => dlc,
+            None => return Err(CannotRollover::NoDlc),
+        };
```
Comment (klochowicz) [inline model/src/cfd.rs]: perhaps neater:

```suggestion
        let dlc = self.dlc.as_ref().ok_or(CannotRollover::NoDlc)?;
```

**Review by klochowicz — APPROVED** [review]

File: model/src/cfd.rs (hunk 1 above)
Comment (luckysori) [reply model/src/cfd.rs]: Nope, I just forgot!

File: model/src/cfd.rs (hunk 2 above)
Comment (luckysori) [reply model/src/cfd.rs]: Yep, for sure!

**Review by da-kami — APPROVED** [review]

**Pull request merged**